src/asmcnc/apps/wifi_app/screen_wifi.py

```diff
@@ -455,6 +463,18 @@ def check_credentials(self):
         else: 
             self.connect_wifi()
 
+    def is_wlan0_connected(self):
+        #returns "state UP" or "state DOWN" depending on whether wlan0 is connected or not
+        state_raw = os.popen('ip addr show | grep "wlan0" | grep -oP "state\s\w+"').read()
+        state = state_raw.split(" ")[1].strip("\n")
+
+        if state == "UP":

```

Comment:
You can simplify this block to something like: 

print("wlan0 connection status: " + str(state))
return state == "UP"


>>> def func(arg): return arg==1
>>> func(0)
False
>>> func(1)
True